flocker/node/agents/cinder.py

```diff
@@ -539,15 +539,16 @@ def attach_volume(self, blockdevice_id, attach_to):
         return attached_volume
 
     def detach_volume(self, blockdevice_id):
-        our_id = self.compute_instance_id()
         try:
             cinder_volume = self.cinder_volume_manager.get(blockdevice_id)
         except CinderNotFound:
             raise UnknownVolume(blockdevice_id)
+        server_id = _blockdevicevolume_from_cinder_volume(
```

Comment:
:grey_question: I'm surprised this change didn't break any cinder functional tests.

Do you think there would be any value at testing this at the functional level? Perhaps it's not worth setting up the two instances to run the functional test... but it'd be nice to have some coverage here.